app/assets/javascripts/discourse/app/models/post-stream.js

```diff
@@ -617,27 +628,46 @@ export default RestModel.extend({
     }
 
     const loadedAllPosts = this.loadedAllPosts;
+    this._loadingPostIds = this._loadingPostIds || [];
 
-    if (this.stream.indexOf(postId) === -1) {
-      this.stream.addObject(postId);
-      if (loadedAllPosts) {
-        this.set("loadingLastPost", true);
-        return this.findPostsByIds([postId])
-          .then((posts) => {
-            const ignoredUsers =
-              User.current() && User.current().get("ignored_users");
-            posts.forEach((p) => {
-              if (ignoredUsers && ignoredUsers.includes(p.username)) {
-                this.stream.removeObject(postId);
-                return;
-              }
-              this.appendPost(p);
-            });
-          })
-          .finally(() => {
-            this.set("loadingLastPost", false);
-          });
+    let missingIds = [];
+
+    postIds.forEach((postId) => {
+      if (postId && this.stream.indexOf(postId) === -1) {
+        missingIds.push(postId);
       }
+    });
+
+    if (missingIds.length === 0) {
+      return resolved;
+    }
+
+    if (loadedAllPosts) {
+      missingIds.forEach((postId) => {
+        if (this._loadingPostIds.indexOf(postId) === -1) {
+          this._loadingPostIds.push(postId);
+        }
+      });
+      this.set("loadingLastPost", true);
+      return this.findPostsByIds(this._loadingPostIds, opts)
+        .then((posts) => {
+          this._loadingPostIds = null;
+          const ignoredUsers =
+            User.current() && User.current().get("ignored_users");
+          posts.forEach((p) => {
+            if (ignoredUsers && ignoredUsers.includes(p.username)) {
+              this.stream.removeObject(p.id);
+              return;
+            }
+            this.stream.addObject(p.id);
+            this.appendPost(p);
+          });
+        })
+        .finally(() => {
+          this.set("loadingLastPost", false);
```

Comment:
Hmm, loadingLastPost gets set to false in-between retries? is that going to flicker the display at all?

Reply:
From my testing, the flicker and weird all happens when you have stuff in the post stream which you have not loaded. loading last post does not cause too much issue and flicker.